File: solutions/python/bob/1/bob.py
def response(hey_bob):
    hey_bob = hey_bob.strip()
    if not hey_bob:
        return "Fine. Be that way!"
    
    is_question = hey_bob.endswith('?')
    is_shelling = hey_bob.isupper()
    
    if is_shelling and is_question:
        return "Calm down, I know what I'm doing!"
    if is_shelling:
        return "Whoa, chill out!"
    if is_question:
        return "Sure."
    
    return "Whatever."

File: solutions/ruby/bob/1/bob.rb
class Bob
  def self.hey(remark)
    remark = remark.strip
    if remark.empty?
      "Fine. Be that way!"
    elsif remark.end_with?("?")
      if remark.upcase == remark && remark.downcase != remark
        "Calm down, I know what I'm doing!"
      else
        "Sure."
      end
    elsif remark.upcase == remark && remark.downcase != remark
      "Whoa, chill out!"
    else
      "Whatever."
    end
  end
end
